microceph/ceph/replication_rbd.go

```diff
@@ -400,24 +450,53 @@ func handleImageDisablement(rh *RbdReplicationHandler) error {
 	return configureImageMirroring(rh.Request)
 }
 
-// enableImageFeatures enables the list of rbd features on the requested resource.
-func enableRbdImageFeatures(poolName string, imageName string, features []string) error {
-	for _, feature := range features {
-		err := configureImageFeatures(poolName, imageName, "enable", feature)
-		if err != nil && !strings.Contains(err.Error(), "one or more requested features are already enabled") {
-			return err
-		}
+// handlePoolDePro verifies the pool state/ peers and
+func handlePoolDePro(rh *RbdReplicationHandler, poolName string) error {
```

Comment:
I feel it would be clearer if we had a specific function for promoting and demoting, also easier to troubleshoot

Reply:
ack

Reply:
have fixed the handling of promotion/demotion.